PluginsAndFeatures/azure-toolkit-for-intellij/rider/src/org/jetbrains/plugins/azure/functions/AzureCoreToolsMissingNupkgInstaller.kt

```diff
@@ -52,8 +53,8 @@ class AzureCoreToolsMissingNupkgInstaller : StartupActivity {
                 file.extension.equals("vb", true) ||
                 file.extension.equals("fs", true)
 
-        private fun existsInCurrentProject(projectFileIndex: ProjectFileIndex, file: VirtualFile): Boolean =
-                file.exists() && projectFileIndex.isInContent(file)
+        private fun existsInCurrentProject(project: Project, workspaceModel: WorkspaceModel, file: VirtualFile): Boolean =
+                file.exists() && VfsUtil.isAncestor(VfsUtil.findFileByIoFile(File(project.basePath!!), false)!!, file, false)
```

Comment:
Would not it be more clear to check with this API?:
```
workspaceModel.getProjectModelEntities(file, project).isNotEmpty()
```
I'm a bit concerned about NPEs with number of unwraps :)

Reply:
Unfortunately, `workspaceModel.getProjectModelEntities(file, project).isNotEmpty()` always returns `false` in these cases. The model only gets updated later.

Reply:
This code would not work in general. 
The proper way: `WorkspaceProjectRootsTracker.getInstance(project).contains(file)`